react-carousel-component/my-app/src/Carousel.js

```diff
@@ -0,0 +1,73 @@
+import { useEffect, useState } from 'react';
+import './Carousel.css';
+import { AiOutlineArrowLeft } from 'react-icons/ai';
+import { AiOutlineArrowRight } from 'react-icons/ai';
+import { FaCircle } from 'react-icons/fa';
+import { FaRegCircle } from 'react-icons/fa';
+
+export default function Carousel({ pokedex }) {
+  const [current, setCurrent] = useState(0);
+
+
+  useEffect(() => {
+    function autoSwap() {
+      incrementIndex();
+    }
+    const autoRotate = setInterval(autoSwap, 3000)
+    return () => {
+      clearInterval(autoRotate);
+    }
+  });
```

Comment:
You should set `incrementIndex` as a dependency, but that requires using `useCallback, which we haven't studied yet. I'm not convinced the `autoSwap` hack you use here will work. But in this particular case, not having any dependencies in `useEffect` works.